src/components/profile-tab.js

```diff
@@ -0,0 +1,146 @@
+import by from 'thenby';
+import gravatar from 'gravatar';
+import memoizee from 'memoizee';
+import React, { useState } from 'react';
+import Logger from '../common/logger.js';
+import localeCompare from 'locale-compare';
+import { useUser } from './user-context.js';
+import errorHelper from '../ui/error-helper.js';
+import { useService } from './container-context.js';
+import { useLanguage } from './language-context.js';
+import { Trans, useTranslation } from 'react-i18next';
+import UserApiClient from '../services/user-api-client.js';
+import CountryNameProvider from '../data/country-name-provider.js';
+import CountryFlagAndName from './localization/country-flag-and-name.js';
+import { Form, Input, Alert, Avatar, Button, Select, message } from 'antd';
+import { formItemLayoutShape, tailFormItemLayoutShape } from '../ui/default-prop-types.js';
+
+const logger = new Logger(import.meta.url);
+
+const FormItem = Form.Item;
+const Option = Select.Option;
+
+const AVATAR_SIZE = 256;
+
+const createCountryNames = memoizee((countryNameProvider, language) => {
+  return Object.entries(countryNameProvider.getData(language))
+    .map(([key, name]) => ({ key, name }))
+    .sort(by(x => x.name, { cmp: localeCompare(language) }));
+}, { max: 1 });
+
+function ProfileTab({ formItemLayout, tailFormItemLayout }) {
+  const user = useUser();
+  const { language } = useLanguage();
+  const { t } = useTranslation('profileTab');
+  const userApiClient = useService(UserApiClient);
+  const countryNameProvider = useService(CountryNameProvider);
+
+  const profile = user.profile || { country: '' };
+  const gravatarUrl = gravatar.url(user.email, { s: AVATAR_SIZE, d: 'mp' });
+  const gravatarRagistrationUrl = `https://${language}.gravatar.com/`;
```

Comment:
Maybe we could move this to the urls package? urls.getGravatarRegistrationUrl(language)?

Reply:
To me that's for composing internal/website URLs, I would not put this one there.